proxy/clientcertvalidate/subject.go

```diff
@@ -0,0 +1,197 @@
+func (ccs *defaultParsedSubject) WithKVs(field string, values []string) {
+	ccs.kvs[field] = values
+}
+func (ccs *defaultParsedSubject) WithRegexpKVs(field string, values []*regexp.Regexp) {
+	ccs.regexpkvs[field] = values
+}
+
+func (ccs *defaultParsedSubject) X509Validate(cert *x509.Certificate) error {
+	switch ccs.inputValuesType {
+	case ClientCertificateSubjectPrefixString:
+		for k, v := range ccs.kvs {
+			switch k {
+			case clientCertSubjectCommonName:
+				expected := []string{cert.Subject.CommonName}
+				if err := testCertValuesSlicesStringSubject(clientCertSubjectCommonName, v, expected); err != nil {
+					return err
+				}
+			case clientCertSubjectCountry:
+				if err := testCertValuesSlicesStringSubject(clientCertSubjectCountry, v, cert.Subject.Country); err != nil {
+					return err
+				}
+			case clientCertSubjectProvince:
+				if err := testCertValuesSlicesStringSubject(clientCertSubjectProvince, v, cert.Subject.Province); err != nil {
+					return err
+				}
+			case clientCertSubjectLocality:
+				if err := testCertValuesSlicesStringSubject(clientCertSubjectLocality, v, cert.Subject.Locality); err != nil {
+					return err
+				}
+			case clientCertSubjectOrganization:
+				if err := testCertValuesSlicesStringSubject(clientCertSubjectOrganization, v, cert.Subject.Organization); err != nil {
+					return err
+				}
+			case clientCertSubjectOrganizationalUnit:
+				if err := testCertValuesSlicesStringSubject(clientCertSubjectOrganizationalUnit, v, cert.Subject.OrganizationalUnit); err != nil {
+					return err
+				}
+			}
+		}
+	case ClientCertificateSubjectPrefixPattern:
+		for k, v := range ccs.regexpkvs {
+			switch k {
+			case clientCertSubjectCommonName:
+				expected := []string{cert.Subject.CommonName}
+				if err := testCertValuesSlicesRegexpSubject(clientCertSubjectCommonName, v, expected); err != nil {
+					return err
+				}
+			case clientCertSubjectCountry:
+				if err := testCertValuesSlicesRegexpSubject(clientCertSubjectCountry, v, cert.Subject.Country); err != nil {
+					return err
+				}
+			case clientCertSubjectProvince:
+				if err := testCertValuesSlicesRegexpSubject(clientCertSubjectProvince, v, cert.Subject.Province); err != nil {
+					return err
+				}
+			case clientCertSubjectLocality:
+				if err := testCertValuesSlicesRegexpSubject(clientCertSubjectLocality, v, cert.Subject.Locality); err != nil {
+					return err
+				}
+			case clientCertSubjectOrganization:
+				if err := testCertValuesSlicesRegexpSubject(clientCertSubjectOrganization, v, cert.Subject.Organization); err != nil {
+					return err
+				}
+			case clientCertSubjectOrganizationalUnit:
+				if err := testCertValuesSlicesRegexpSubject(clientCertSubjectOrganizationalUnit, v, cert.Subject.OrganizationalUnit); err != nil {
+					return err
+				}
+			}
+		}
+	default:
+		return errors.New("unknown subject header")
+	}
+
+	return nil
+}
+
+func testCertValuesSlicesStringSubject(field string, expected, certValues []string) error {
```

Comment:
field is not used, clientCertSubjectOrganizationalUnit is hard coded

Reply:
Thanks, fixed: https://github.com/grepplabs/kafka-proxy/pull/74/commits/0ff634d1ff1902a75b36704db539b2c76cdd95d7.